Add page about official OpenTofu Docker image

nikitabarskov opened the pull request:
Documentation website was lacking a mention of official OpenTofu Docker image.
So I've added it.

See https://opentofucommunity.slack.com/archives/C05R0FD0VRU/p1708027559842219

janosdebugs commented on website/docs/intro/install/docker.mdx:
```diff
@@ -0,0 +1,47 @@
+
+# Use OpenTofu as Docker Image
+
+OpenTofu is available as [Docker images](https://github.com/opentofu/opentofu/pkgs/container/opentofu),
```
```suggestion
OpenTofu is available as [OCI container images](https://github.com/opentofu/opentofu/pkgs/container/opentofu),
```

janosdebugs commented on website/docs/intro/install/index.mdx:
```diff
@@ -54,5 +54,12 @@ You can install OpenTofu via a wide range of methods. Please select your operati
+            href: "/docs/intro/install/docker",
+            label: "Docker Image",
+            description:
+                "Use official Docker Image available on GitHub Container Registry.",
```
```suggestion
                "Use official container image available on GitHub Container Registry.",
```

janosdebugs commented on website/docs/intro/install/index.mdx:
```diff
@@ -54,5 +54,12 @@ You can install OpenTofu via a wide range of methods. Please select your operati
+        {
+            type: "link",
+            href: "/docs/intro/install/docker",
+            label: "Docker Image",
```
```suggestion
            label: "Container Image",
```

janosdebugs commented on website/docs/intro/install/docker.mdx:
```diff
@@ -0,0 +1,47 @@
+sidebar_position: 99
+sidebar_label: Docker Image
+description: |-
+  Use official Docker Image available on GitHub Container Registry.
```
```suggestion
  Use official container image available on the GitHub Container Registry.
```

janosdebugs commented on website/docs/intro/install/docker.mdx:
```diff
@@ -0,0 +1,47 @@
+---
+sidebar_position: 99
+sidebar_label: Docker Image
```
```suggestion
sidebar_label: Container Image
```

janosdebugs commented on website/docs/intro/install/docker.mdx:
```diff
@@ -0,0 +1,47 @@
+
+## Usage
+
+To be able to pull the image generate a PAT with scope `packages:read` and then
+use it to [login][ghcr-login] to GitHub Packages registry.
```
This is incorrect, you can pull without logging in, but rate limits may apply (see suggestion below).

```suggestion
```

janosdebugs commented on website/docs/intro/install/docker.mdx:
```diff
@@ -0,0 +1,47 @@
+
+<CodeBlock language="bash">{DockerRunScript}</CodeBlock>
+
+[ghcr-login]: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry
```
```suggestion
:::tip

If you run into rate limiting issues with GitHub, create a [Personal Access Token](https://github.com/settings/tokens) and [log in to the registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry).

:::
```

janosdebugs commented on website/docs/intro/install/docker-pull.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# Copyright (c) The OpenTofu Authors
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (c) 2023 HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+
```
Since these go into an example, the comments make the snippet rather lengthy. I would suggest removing both the hashbang line and the copyright headers.

![image](https://github.com/opentofu/opentofu/assets/86970079/076375c5-70b7-4bb0-95c8-32d944f21d08)


```suggestion
```

janosdebugs commented on website/docs/intro/install/docker-run.sh:
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+# Copyright (c) The OpenTofu Authors
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (c) 2023 HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+
```
Same as above.

```suggestion
```

janosdebugs reviewed (approved):
Hey @nikitabarskov thank you so much for setting this up! Happy to approve as-is, just a few minor nit-picks that can possibly be improved (feel free to reject).

nikitabarskov commented:
> Hey @nikitabarskov thank you so much for setting this up! Happy to approve as-is, just a few minor nit-picks that can possibly be improved (feel free to reject).

Hey @janosdebugs, I've addressed all the comment, feel free to re-review and we can merge then!
Thank you!

janosdebugs commented:
Hi @nikitabarskov thank you very much.

As it turns out, [we already have multiarch images](https://github.com/opentofu/opentofu/issues/1273#issuecomment-1950318939), so you could simply pull the `1.6.1` tag. Would you want to address this in the PR or should we just merge and address it in a separate PR?

nikitabarskov commented:
> Hi @nikitabarskov thank you very much.
> 
> As it turns out, [we already have multiarch images](https://github.com/opentofu/opentofu/issues/1273#issuecomment-1950318939), so you could simply pull the `1.6.1` tag. Would you want to address this in the PR or should we just merge and address it in a separate PR?

Also updated see the new description :)

janosdebugs commented on website/docs/intro/install/docker.mdx:
```diff
@@ -0,0 +1,48 @@
+
+<CodeBlock language="bash">{DockerRunScript}</CodeBlock>
+
+:::tip
+If you run into rate limiting issues with GitHub, create a
+[Personal Access Token](https://github.com/settings/tokens) and
+[log in](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)
+to the registry.
+:::
```
Just because this may cause issues with Docusaurus, two line breaks:
```suggestion
:::tip

If you run into rate limiting issues with GitHub, create a
[Personal Access Token](https://github.com/settings/tokens) and
[log in](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)
to the registry.

:::
```

janosdebugs reviewed (approved):
LGTM, tiny fix and then it should be ready to merge. Thanks for your work @nikitabarskov !

janosdebugs commented:
@nikitabarskov thanks, merged now, this will need a backport to show up, but we'll take care of this. Thank you for your work!
